sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Repositories/ContainerRepositoryClient.cs

```diff
@@ -0,0 +1,135 @@
+        {
+        }
+
+        /// <summary>
+        /// <param name="endpoint"></param>
+        /// <param name="repository"> Name of the image (including the namespace). </param>
+        /// <param name="username"></param>
+        /// <param name="password"></param>
+        /// <param name="options"></param>
+        /// </summary>
+        public ContainerRepositoryClient(Uri endpoint, string repository, string username, string password, ContainerRegistryClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(endpoint, nameof(repository));
+            Argument.AssertNotNull(endpoint, nameof(username));
+            Argument.AssertNotNull(endpoint, nameof(password));
+            Argument.AssertNotNull(options, nameof(options));
+
+            _pipeline = HttpPipelineBuilder.Build(options, new BasicAuthenticationPolicy(username, password));
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            Endpoint = endpoint;
+            _repository = repository;
+
+            _restClient = new RepositoryRestClient(_clientDiagnostics, _pipeline, Endpoint.AbsoluteUri);
+        }
+
+        /// <summary> Initializes a new instance of RepositoryClient for mocking. </summary>
+        protected ContainerRepositoryClient()
+        {
+        }
+
+        /// <summary> Get repository properties. </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<RepositoryProperties>> GetPropertiesAsync(CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRepositoryClient)}.{nameof(GetProperties)}");
+            scope.Start();
+            try
+            {
+                return await _restClient.GetAttributesAsync(_repository, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Get repository properties. </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual Response<RepositoryProperties> GetProperties(CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRepositoryClient)}.{nameof(GetProperties)}");
+            scope.Start();
+            try
+            {
+                return _restClient.GetAttributes(_repository, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Update the attribute identified by `name` where `reference` is the name of the repository. </summary>
```

Comment:
It's not clear what `name` and `reference` are referring to here. Is there some more context we could add?